api/src/exam-environment/routes/exam-environment.test.ts

```diff
@@ -260,8 +269,30 @@ describe('/exam-environment/', () => {
         expect(res.status).toBe(404);
       });
 
-      xit('should return an error if the exam prerequisites are not met', async () => {
-        // TODO: Waiting on prerequisites
+      it('should return an error if the exam prerequisites are not met', async () => {
+        await fastifyTestInstance.prisma.user.update({
+          where: { id: defaultUserId },
+          data: {
+            completedChallenges: []
+          }
+        });
+
+        const body: Static<typeof examEnvironmentPostExamGeneratedExam.body> = {
+          examId: mock.exam.id
+        };
+        const res = await superPost('/exam-environment/exam/generated-exam')
+          .send(body)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res.body).toStrictEqual({
+          code: 'FCC_EINVAL_EXAM_ENVIRONMENT_PREREQUISITES',
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+          message: expect.any(String)
```

Comment:
```suggestion
          message: 'User has not completed prerequisites.'
```
Again, minor, but we know the message, so we can just use it.

Reply:
I do mention elsewhere in the tests that we do not necessarily want to test the exact string returned here, because it is not a guaranteed part of the api - it is an additional explanation.

So, a changing `code` is a breaking change, but the `message`s can change anyhow we see fit. E.g. in this case, I expect to change this to add additional information like `User has not completed challenges "a", "b", "c".`

Reply:
Makes sense. I would still probably make it strict once you've settled on `User has not completed challenges "a", "b", "c".`, though. At that point it's unlikely to change very often.